web/components/groups/group-overview.tsx

```diff
@@ -0,0 +1,372 @@
+        group={group}
+        creator={creator}
+        isEditable={isEditable}
+        user={user}
+        memberIds={memberIds}
+      />
+    </Col>
+  )
+}
+
+function GroupOverviewPinned(props: {
+  group: Group
+  posts: Post[]
+  isEditable: boolean
+}) {
+  const { group, posts, isEditable } = props
+  const [pinned, setPinned] = useState<JSX.Element[]>([])
+  const [open, setOpen] = useState(false)
+  const [editMode, setEditMode] = useState(false)
+
+  useEffect(() => {
+    async function getPinned() {
+      if (group.pinnedItems == null) {
+        updateGroup(group, { pinnedItems: [] })
+      } else {
+        const itemComponents = await Promise.all(
+          group.pinnedItems.map(async (element) => {
+            if (element.type === 'post') {
+              const post = await getPost(element.itemId)
+              if (post) {
+                return <PostCard post={post as Post} />
+              }
+            } else if (element.type === 'contract') {
+              const contract = await getContractFromId(element.itemId)
+              if (contract) {
+                return <ContractCard contract={contract as Contract} />
+              }
+            }
+          })
+        )
+        setPinned(
+          itemComponents.filter(
+            (element) => element != undefined
+          ) as JSX.Element[]
+        )
+      }
+    }
+    getPinned()
+  }, [group, group.pinnedItems])
+
+  async function onSubmit(selectedItems: { itemId: string; type: string }[]) {
+    await updateGroup(group, {
+      pinnedItems: [
+        ...group.pinnedItems,
+        ...(selectedItems as { itemId: string; type: 'contract' | 'post' }[]),
+      ],
+    })
+    setOpen(false)
+  }
+
+  return isEditable || pinned.length > 0 ? (
+    <>
+      <Row className="mb-3 items-center justify-between">
+        <SectionHeader label={'Pinned'} />
+        {isEditable && (
+          <Button
+            color="gray"
+            size="xs"
+            onClick={() => {
+              setEditMode(!editMode)
+            }}
+          >
+            {editMode ? (
+              'Done'
+            ) : (
+              <>
+                <PencilIcon className="inline h-4 w-4" />
+                Edit
+              </>
+            )}
+          </Button>
+        )}
+      </Row>
+      <div>
+        <Masonry
+          breakpointCols={{ default: 2, 768: 1 }}
+          className="-ml-4 flex w-auto"
+          columnClassName="pl-4 bg-clip-padding"
+        >
+          {pinned.length == 0 && !editMode && (
+            <div className="flex flex-col items-center justify-center">
+              <p className="text-center text-gray-400">
+                No pinned items yet. Click the edit button to add some!
+              </p>
+            </div>
+          )}
+          {pinned.map((element, index) => (
+            <div className="relative my-2">
+              {element}
+
+              {editMode && (
+                <div>
```

Comment:
You could extract this 'X' button to a small component defined later in the file.

More small components is generally good style. It reduces nesting and makes the parent component & child component more readable.